### ISSUE TITLE
Would be nice to have Kubernetes Pod and Service Spec

### ISSUE THREAD

**Issue by uruddarraju** [sdelements/lets-chat#551]
You guys have done a  great work with this piece, it would be very nice if we can have a pod spec (kubernetes) to deploy it in a kubernetes cluster for people using kubernetes instead of docker compose/swarm.

**Comment by hhaidar** [comment]
@uruddarraju thank you for the kind words! Is it something you can pick up?

**Comment by uruddarraju** [comment]
yeah I would love to.

**Comment by rawlingsj** [comment]
We have an OpenShift version of Lets Chat Kubernetes config here http://central.maven.org/maven2/io/fabric8/jube/images/fabric8/letschat/2.2.5/letschat-2.2.5-kubernetes.json

We've been using Lets Chat on [fabric8](http://fabric8.io/) which runs on OpenShift and / or Kubernetes.  We're working on a few CD and DevOps demos over the next couple of weeks which show how we've integrated Lets Chat combined with hubot and a few other equally awesome OSS projects.

The Lets Chat image we're using however has been [slightly customised](https://github.com/fabric8io/lets-chat/commits/master) until we sort our single sign for the platform.  Also the JSON I've linked to is OpenShift specific, it uses a Template Kind but we have tooling to convert it into List and use the parameters at the bottom of the JSON replace some of the env var values.  That is all that's needed to run on Kubernetes if you wanted to hack that one.

If you like tomorrow I'd happily switch it and contribute a non customised, Kubernetes only config for you and others.  We've not mounted the mongodb to a persistent volume yet as we're only in testing at the moment but it is on the list.

If you're interested on finding out more about [fabric8](http://fabric8.io/) (shameless plug sorry) there's a 5 min getting started which shows how to get openshift which includes Kubernetes running in a vagrant image, here's it's easy to run the Lets Chat pod (we are working hard to reproduce all this on plain old Kubernetes very soon) https://medium.com/fabric8-io/getting-started-with-fabric8-7bd7ff14958e

In short fabric8 amonst other things is developing a DevOps platform and Lets Chat is an awesome integration we have been using.  For info about why we use Lets Chat and think it is so important https://medium.com/fabric8-io/fabric8-s-social-side-7a08ec38d76d
